src/main/java/com/example/tournamentMaker/randomResultGenerator/RandomResultService.java

```diff
@@ -32,35 +34,35 @@ class RandomResultService {
     private final TeamRepository teamRepository;
     private static final int MIN_YELLOW_CARDS_IN_MATCH = 0;
     private static final int MAX_YELLOW_CARDS_IN_MATCH = 3;
-    private static final int ZERO_PERCENT = 0;
-    private static final int ONE_HUNDRED_PERCENT = 100;
+    private final Logger logger = LoggerFactory.getLogger(RandomResultService.class);
 
-    public void drawRoundsResults(RandomResultRequest request) {
+    public void drawLotRoundsResults(RandomResultRequest request) {
         Tournament tournament = tournamentRepository.findByName(request.getTournamentName())
                 .orElseThrow(() -> {
                     throw new NoSuchElementException(Constans.NO_TOURNAMENT_FOUND);
                 });
-        String[] drawRounds = request.getRoundsToDraw().split("-");
-        int firstRound = Integer.parseInt(drawRounds[0]);
-        int lastRound = Integer.parseInt(drawRounds[1]);
+        String[] rounds = request.getRoundsToDraw().split("-");
+        int firstRound = Integer.parseInt(rounds[0]);
+        int lastRound = Integer.parseInt(rounds[1]);
 
-        Sport sport = tournament.getSport();
-        switch (sport) {
-            case FOOTBALL -> drawFootballRounds(firstRound, lastRound, tournament);
-        }
+        drawLotFootballRounds(firstRound, lastRound, tournament);
     }
 
-    private void drawFootballRounds(int firstRound, int lastRound, Tournament tournament) {
-        for (int i = firstRound; i <= lastRound; i++) {
-            drawFootballRoundResult(i - 1, tournament);
-        }
+    private void drawLotFootballRounds(int firstRound, int lastRound, Tournament tournament) {
+        IntStream.rangeClosed(firstRound - 1, lastRound - 1).forEach(roundNumber -> {
+            try {
+                Round round = tournament.getRounds().get(roundNumber);
+                drawLotFootballRoundResult(round);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                logger.error("There is no round with index " + (roundNumber - 1));
```

Comment:
tyle wystarczy ? 
Co sie stanie jak poleci wyjatek ? Jakie to konsekwencje bedzie mialo dla aplikacji ? Co sie potem powinno stac od strony usera?